invenio_stats/aggregations.py

```diff
@@ -12,21 +12,185 @@
 
 import datetime
 from collections import OrderedDict
+from copy import deepcopy
 
 import six
 from dateutil import parser
+from elasticsearch import VERSION as ES_VERSION
 from elasticsearch.helpers import bulk
 from elasticsearch_dsl import Index, Search
 from invenio_search import current_search_client
 
 from .utils import get_size
+from .utils import get_doctype, get-size
+
+SUPPORTED_INTERVAL = OrderedDict([
+    ('hour', '%Y-%m-%dT%H'),
+    ('day', '%Y-%m-%d'),
+    ('month', '%Y-%m'),
+    ('year', '%Y')
+])
 
 
 def filter_robots(query):
     """Modify an elasticsearch query so that robot events are filtered out."""
     return query.filter('term', is_robot=False)
 
 
+def format_range_dt(d, interval):
```

Comment:
I feel like I've seen this somewhere else... (could be Zenodo as well :see_no_evil:)